### DISCUSSION TITLE
Unable to connect Device to MQTT

### DISCUSSION THREAD

**Discussion by Freddiebe** [john30/ebusd#902] [General]
Hi,

I'm unsure that this is technically an issue, as it's probably more my lack of knowledge / experience. Anyway, I have just received my new adapter (thank you very much), and have flashed it to the newest firmware:

![image](https://github.com/john30/ebusd/assets/134110254/1e42fde9-c840-4502-9850-77c9e0556e4f)

I have installed the latest ebusd (MQTT version) on a Raspberry Pi 3 running Busllseye. I have changed /etc/default/ebusd to point to the new device with the following line:

EBUSD_OPTS="--scanconfig -d ens:192.168.1.176:9999"

This appears to be working as I can now read items on the bus:

![image](https://github.com/john30/ebusd/assets/134110254/636e58b8-6737-4f17-b03e-f1ef5de5dd71)

I have Mosquitto installed. If I try to publish device data from ebusd (for use in home assistant), no data is sent other than ebusd uptime and status. I get the following errors for the device / sensor data:

![image](https://github.com/john30/ebusd/assets/134110254/f2e751ea-bf75-4311-9275-36b19bcbf6a5)

I appreciate that I am probably doing something wrong, or have missed out a step, but any pointers you could give me would be greatly appreciated.

By the way, I'm trying to interface with a Vaillant Arotherm Plus 5kW ASHP with VRC720 .

Many thanks,

Fred

**Comment by Freddiebe** [comment]
I managed to get this working by rebooting the ebus adapter. If the adapter is running before starting ebusd, then it dosn't work, but the other way round, everything is fine.

**Reply by QubaB** [reply]
I have had similar problem, I need to restart ebus adapter to be recognized by ebusd - see thread #890 . At the end the problem was that my computer had 2 IP addresses. 

When your ebusd daemon is stopped and you restart your ebus adapter. Does it show "ebusd connected: no"?  My adapter showed that it is connected.
